synchronizer/time_provider.go

```diff
@@ -0,0 +1,19 @@
+package synchronizer
+
+import (
+	"time"
+)
+
+// TimeProvider is a interface for classes that needs time and we want to be able to unittest it
+type TimeProvider interface {
+	// Now returns current time
+	Now() time.Time
+}
+
+// DefaultTimeProvider is the default implementation of TimeProvider
+type DefaultTimeProvider struct{}
+
+// Now returns current time
+func (d DefaultTimeProvider) Now() time.Time {
+	return time.Now()
+}
```

Comment:
I think we could avoid this. I mean I think it doesn't bring any value to the project

Reply:
I haven't write the unittest for the cache, but I'll need that to be able to test it fast an not depend on real time